### PR TITLE
Add bits as an optional feature for bitvec

### PR DESCRIPTION
**Pull request by wcampbell0x2a**
This is like the final boss of all my performance work. I try to explain the "no warranty and testing are given for the disable of this feature", while still implementing enough to make this useful in the case of "I want perf to matter".

image

bench: https://github.com/wcampbell0x2a/deku-bench/pull/24

### CODE REVIEW

**Comment by sharksforarms** [comment]
I think the warning is fair, although, what do we need to do to have more assurance and remove that warning?

**Comment by wcampbell0x2a** [comment]
> I think the warning is fair, although, what needs to be done to have more assurance and remove that warning?

I updated this code to behave with running `cargo test --lib --examples --tests --features std --no-default-features`, and added it to CI. This way I am more confident that this options works.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
       - run: cargo test --all
       # run examples
       - run: cargo run --example 2>&1 | grep -P '   ' | awk '{print $1}' | xargs -i cargo run --example {}
+      # test with no bits feature (don't test docs)
```
Comment (sharksforarms) [inline .github/workflows/main.yml:25]: nice

**Review by sharksforarms — APPROVED** [review]
very nice, thank you

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -15,3 +15,9 @@ jobs:
         with:
           branchName: ${{ github.base_ref }}
           token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: boa-dev/criterion-compare-action@v3
+        with:
+          branchName: ${{ github.base_ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          defaultFeatures: false
```
Comment (wcampbell0x2a) [inline .github/workflows/pull_request.yml:22]: This option doesn't seem to be adding ``--no-default-features`` like it should :/


idk javascript: https://github.com/boa-dev/criterion-compare-action/blob/59f4d964c5f19d7f13d36f5c0944b18ce6652cb0/main.js#L37

Comment (wcampbell0x2a) [reply .github/workflows/pull_request.yml:22]: Maybe I fixed it, i'll try out my test version here...

https://github.com/boa-dev/criterion-compare-action/pull/117

Comment (wcampbell0x2a) [reply .github/workflows/pull_request.yml:22]: Didn't fix it, guess I won't become a js dev anytime soon 😆

**Review by sharksforarms — APPROVED** [review]

**Pull request merged**